### ISSUE TITLE
bug in setDebugDraw 

### ISSUE THREAD

**Issue by alexzheng** [ymobe/rapanui#8]
I run the sample rapanui-samples/physics/rn-physics-lists and is report an error

I look into the source 

--we need the layer from RNScene received
--keep it in mind for future changes
function setDebugDraw(screen)

```
local layerfordebug = screen.layer
len = table.getn(screen.sprites)
for i = 0, len, 1 do
    screen.sprites[i].visible = false
end
layerfordebug:setBox2DWorld(world)
```

end

i should start from 1

**Comment by MattiaFortunati** [comment]
Hi!

The bug occured because we have just changed the lists in groups so we needed to change this list, too.
I fixed this and I fixed also the table list screen.sprites in RNScreen so everything should work now.

Just remember that setDebugDraw(screen) simply sets all the sprites in the list as invisibles, so if we add more sprites after we call it, those new sprites will be visible instead. 

That's because I haven't added the code to check whether the entire world is set to debugDraw, so new sprites will be invisibles but it shouldn't be a problem for debug purposes.

Mattia

**Comment by alexzheng** [comment]
I got a little confusion in the code:
where is the image is added to?
is it added to two different mainGroup?
scene1.lua
function onCreate()
    --add things to mainGroup
    background = RNFactory.createImage("images/background-purple.png");background.x=160;background.y=240;
    print("----------------")
    print(mainGroup)
    print(background)

```
tile1a = RNFactory.createImage("images/tile1.png");tile1a.x=160;tile1a.y=240;
tile1b = RNFactory.createImage("images/tile1.png");tile1b.x=100;tile1b.y=240;
tile1c = RNFactory.createImage("images/tile1.png");tile1c.x=260;tile1c.y=240;

mainGroup:insert(background)
mainGroup:insert(tile1a)
mainGroup:insert(tile1b)
mainGroup:insert(tile1c)
--return mainGroup  
return mainGroup    
```

end

**Comment by MattiaFortunati** [comment]
Hi!

This is image factory, where images are created and put on the screen:

<code>background = RNFactory.createImage("images/background-purple.png");background.x=160;background.y=240;
tile1a = RNFactory.createImage("images/tile1.png");tile1a.x=160;tile1a.y=240;
tile1b = RNFactory.createImage("images/tile1.png");tile1b.x=100;tile1b.y=240;
tile1c = RNFactory.createImage("images/tile1.png");tile1c.x=260;tile1c.y=240;</code>

And here we insert those images in the mainGroup, so we can have a group within the scene handling all the images in it:

<code>mainGroup:insert(background)
mainGroup:insert(tile1a)
mainGroup:insert(tile1b)
mainGroup:insert(tile1c)</code>

So we have only one mainGroup in the scene script.

Let me know if you need better explanations.

Mattia

**Comment by alexzheng** [comment]
hi
the images are inserted into one mainGroup  in RNFactory.lua since no parentGroup is passed  and then inserted into another mainGroup  in scene1.lua. And after translating to scene2, the images are removed in onEnd,but the mainGroup  in RNFactory.lua still reference the images.Is that true?

**Comment by MattiaFortunati** [comment]
Hi!

Yes you were right, they were added to two groups and then removed from only one of them...
That's because while coding scenes and director I forgot that RNFactory adds images in its mainGroup by default.

I fixed some changes in scenes scripts that can resolve this:
We can avoid the default group:insert during factory:

<code>tile1a = RNFactory.createImage("images/tile1.png",{parentGroup=mainGroup});tile1a.x=160;tile1a.y=240;</code>

(like in scene1.lua and scene2.lua)

or we can remove childs from the default group later

<code>for i=1,table.getn(RNFactory.mainGroup.displayObjects),1 do
        RNFactory.mainGroup:removeChild(1);
end</code>

(like for scene3.lua and scene4.lua)

This should remove completely the images from any group or reference.

We are currently working on memory management, so thank you for having discovered this and let me know if you have other problems.

Thank you again ^^

Mattia
